src/main/java/crystal/Storage.java

```diff
@@ -0,0 +1,200 @@
+package crystal;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+
+import crystal.task.Task;
+import crystal.task.Todo;
+import crystal.task.Deadline;
+import crystal.task.Event;
+
+/**
+ * Represents the Storage task.
+ *
+ */
+
+public class Storage {
+    String filepath;
+
+    /**
+     * Constructor for Storage class.
+     *
+     * @param filepath filepath where the list is stored at.
+     *
+     */
+
+    public Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Method to write to the file.
+     *
+     * @param filepath Filepath to store the list at
+     * @param textToAdd Text to be printed on the file
+     *
+     */
+    //Save to file
+    public void writeToFile(String filepath,String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(filepath);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+    /**
+     * Method to save the file.
+     *
+     * @param tasks Tasklist to be saved
+     *
+     */
+    public void saveFile(TaskList tasks) {
+        StringBuilder str = new StringBuilder();
+        for (int i = 0; i < tasks.size(); i++) {
+
+            if (tasks.get(i).toString().startsWith("[T]")) {
+                String stringReplaceT = tasks.get(i).toString().replace("[T]", "T ");
+                if (tasks.get(i).toString().contains("[X]")) {
+                    stringReplaceT = stringReplaceT.replace("[X]", "| 0 |"); //task done
+                    str.append(stringReplaceT + "\n");
+                } else {
+                    stringReplaceT = stringReplaceT.replace("[ ]", "| 1 |");
+                    str.append(stringReplaceT + "\n");
+                }
+
+            } else if (tasks.get(i).toPrint().startsWith("[D]")) {
+                String stringReplaceD = tasks.get(i).toPrint().replace("[D]", "D ");
+                if (tasks.get(i).toPrint().contains("[X]")) {
+                    stringReplaceD = stringReplaceD.replace("[X]", "| 0 |");
+                    stringReplaceD = stringReplaceD.replace("(by:", "|");
+                    stringReplaceD = stringReplaceD.replace(")", "");
+                    str.append(stringReplaceD + "\n");
+                } else {
+                    stringReplaceD = stringReplaceD.replace("[ ]", "| 1 |");
+                    stringReplaceD = stringReplaceD.replace("(by:", "|");
+                    stringReplaceD = stringReplaceD.replace(")", "");
+                    str.append(stringReplaceD + "\n");
+                }
```

Comment:
Seems that this actual string to stored data transformation process is long and repeating... I have noticed that you have created classes for task objects, so, maybe you would like to try including them as class methods, in the aspect of better readability and ease of reuse? :thinking:

Reply:
Ok, i will try that